datalake/src/main/java/com/sequenceiq/datalake/flow/datalake/scale/handler/DatalakeHorizontalScaleCmRolingRestartWaitHandler.java

```diff
@@ -0,0 +1,120 @@
+
+    @Value("${sdx.datalake.scale.sleeptime-sec:10}")
+    private int sleepTimeInSec;
+
+    @Value("${sdx.datalake.scale.timeout-sec:15}")
+    private int timeoutSec;
+
+    @Inject
+    private ClouderaManagerApiFactory clouderaManagerApiFactory;
+
+    @Inject
+    private ClouderaManagerProxiedClientFactory clouderaManagerProxiedClientFactory;
+
+    protected DatalakeHorizontalScaleCmRolingRestartWaitHandler(EventSender eventSender) {
+        super(eventSender);
+    }
+
+    @Override
+    public String selector() {
+        return DATALAKE_HORIZONTAL_SCALE_CM_ROLLING_RESTART_HANDLER.selector();
+    }
+
+    @Override
+    public void accept(Event<DatalakeHorizontalScaleCmRollingRestartRequest> event) {
+        var data = event.getData();
+        var apiClient = clouderaManagerProxiedClientFactory.getProxiedClouderaManagerClient(data.getResourceCrn());
+        var clustersResourceApi = clouderaManagerApiFactory.getClustersResourceApi(apiClient);
+        var commandId = BigDecimal.ZERO;
+        LOGGER.info("CM rolling restart start for {}", data.getResourceName());
+        commandId = executeCMRollingRestart(data, clustersResourceApi);
+        LOGGER.debug("Rolling restart command id: {}", commandId.longValue());
+        pollCM(apiClient, commandId);
+        var finishedEvent = new DatalakeHorizontalScaleFinishedEvent(
+                DATALAKE_HORIZONTAL_SCALE_FINISHED_EVENT.selector(),
+                data.getResourceId(),
+                data.getResourceName(),
+                data.getUserId(),
+                commandId
+        );
+        eventSender().sendEvent(finishedEvent, event.getHeaders());
+    }
+
+    private BigDecimal executeCMRollingRestart(DatalakeHorizontalScaleCmRollingRestartRequest data, ClustersResourceApi clustersResourceApi) {
+        BigDecimal commandId;
+        try {
+            ApiCommand apiCommand = clustersResourceApi.refresh(data.getResourceName());
+            commandId = apiCommand.getId();
+        } catch (ApiException e) {
+            LOGGER.error("CM rolling restart api call failed under the datalake horizontal scale with error msg: {}", e.getMessage());
+            throw new RuntimeException(e);
+        }
+        return commandId;
+    }
+
+    private void pollCM(ApiClient apiClient, BigDecimal commandId) {
+        LOGGER.info("Start CM poll for rolling restart under the datalake horizontal scale");
+        CommandsResourceApi commandsResourceApi = clouderaManagerApiFactory.getCommandsResourceApi(apiClient);
+        Instant timeoutLimit = Instant.now().plusSeconds(timeoutSec);
+        while (isRunning(commandId, commandsResourceApi) && !isTimedout(timeoutLimit)) {
+            try {
+                sleep(sleepTimeInSec * CONVERT_TO_MILLISEC);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private boolean isTimedout(Instant timeoutLimit) {
```

Comment:
this method never returns `true` so there is no need to indicate the return type. Using it in condition will never work correctly